hashin.py

```diff
@@ -157,7 +160,10 @@ def run_single_package(
     package = data["package"]
 
     maybe_restriction = "" if not restriction else "; {0}".format(restriction)
-    new_lines = "{0}=={1}{2} \\\n".format(package, data["version"], maybe_restriction)
+    name = package
+    if req.extras:
+        name += "[{0}]".format(",".join(sorted(req.extras)))
```

Comment:
I think you can probably use the fact that casting the `Requirement` to a string gives you what you're looking for here:

```
>>> from packaging.requirements import Requirement
>>> req = Requirement("requests[security]")
>>> str(req)
'requests[security]'
```

Reply:
wat?! Is it not a standard `set`? I'll take it.

Reply:
Oh snap! I read that wrong. Ignore me.